addons/html_editor/static/src/main/table/table_ui_plugin.js

```diff
@@ -11,25 +11,30 @@ import { TablePicker } from "./table_picker";
  */
 export class TableUIPlugin extends Plugin {
     static name = "table_ui";
-    static dependencies = ["overlay", "table"];
+    static dependencies = ["history", "overlay", "table", "user_command"];
     resources = {
-        powerboxItems: [
+        user_commands: [
             {
-                id: "table",
-                name: _t("Table"),
+                id: "openPickerOrInsertTable",
+                label: _t("Table"),
                 description: _t("Insert a table"),
+                icon: "fa-table",
+                run: this.openPickerOrInsertTable.bind(this),
+            },
+            {
+                id: "openTablePicker",
+                label: _t("Table"),
+                icon: "fa-table",
+                run: this.openPicker.bind(this),
+            },
```

Comment:
I went for B because, as we briefly discussed last week, this plugin is dedicated to the table UI, being separate from the table plugin.
The table plugin does include a "insertTable" user command.